encoding/kzg/prover/prover.go

```diff
@@ -335,3 +349,98 @@ func toUint64Array(chunkIndices []encoding.ChunkNumber) []uint64 {
 	}
 	return res
 }
+
+func (p *Prover) newProver(params encoding.EncodingParams) (*ParametrizedProver, error) {
+	if err := encoding.ValidateEncodingParams(params, p.KzgConfig.SRSOrder); err != nil {
+		return nil, err
+	}
+
+	// Create FFT settings based on params
+	n := uint8(math.Log2(float64(params.NumEvaluations())))
+	if params.ChunkLength == 1 {
+		n = uint8(math.Log2(float64(2 * params.NumChunks)))
+	}
+	fs := fft.NewFFTSettings(n)
+
+	// Create base KZG settings
+	ks, err := kzg.NewKZGSettings(fs, p.Srs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create KZG settings: %w", err)
+	}
+
+	switch p.Config.BackendType {
+	case encoding.GnarkBackend:
+		return p.createGnarkBackendProver(params, fs, ks)
+	case encoding.IcicleBackend:
+		return p.createIcicleBackendProver(params, fs, ks)
+	default:
+		return nil, fmt.Errorf("unsupported backend type: %v", p.Config.BackendType)
+	}
+
+}
+
+func (p *Prover) createGnarkBackendProver(params encoding.EncodingParams, fs *fft.FFTSettings, ks *kzg.KZGSettings) (*ParametrizedProver, error) {
+	if p.Config.GPUEnable {
+		return nil, fmt.Errorf("GPU is not supported in default backend")
+	}
+
+	_, fftPointsT, err := p.SetupFFTPoints(params)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create subgroup FFT settings
+	t := uint8(math.Log2(float64(2 * params.NumChunks)))
+	sfs := fft.NewFFTSettings(t)
+
+	// Set KZG Prover default backend
+	multiproofBackend := &KzgMultiProofGnarkBackend{
+		Fs:         fs,
+		FFTPointsT: fftPointsT,
+		SFs:        sfs,
+		KzgConfig:  p.KzgConfig,
+	}
+
+	// Set KZG Commitments default backend
+	commitmentsBckend := &KzgCommitmentsGnarkBackend{
+		Srs:        p.Srs,
+		G2Trailing: p.G2Trailing,
+		KzgConfig:  p.KzgConfig,
+	}
+
+	return &ParametrizedProver{
+		Encoder:               p.encoder,
+		EncodingParams:        params,
+		KzgConfig:             p.KzgConfig,
+		Ks:                    ks,
+		KzgMultiProofBackend:  multiproofBackend,
+		KzgCommitmentsBackend: commitmentsBckend,
+	}, nil
+}
+
+func (p *Prover) createIcicleBackendProver(params encoding.EncodingParams, fs *fft.FFTSettings, ks *kzg.KZGSettings) (*ParametrizedProver, error) {
+	return CreateIcicleBackendProver(p, params, fs, ks)
+}
+
+// Helper methods for setup
+func (p *Prover) SetupFFTPoints(params encoding.EncodingParams) ([][]bn254.G1Affine, [][]bn254.G1Affine, error) {
+	subTable, err := NewSRSTable(p.KzgConfig.CacheDir, p.Srs.G1, p.KzgConfig.NumWorker)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create SRS table: %w", err)
+	}
+
+	fftPoints, err := subTable.GetSubTables(params.NumChunks, params.ChunkLength)
```

Comment:
I don't think we ever need fftPoints, chance for optimization

Reply:
Will include this in followup optimizations